acceptance-tests/readme.md

```diff
@@ -0,0 +1,100 @@
+# Acceptance Tests
+
+This module contains the acceptance tests for the Bitbucket Server integration plugin, including the UI tests.
+
+## Running the tests
+
+The Jenkins [Acceptance Test Harness framework](https://github.com/jenkinsci/acceptance-test-harness) runs an instance 
+of Jenkins per test and installs the plugin under test using the `.hpi` file in the `../target` folder, defined using 
+the `LOCAL_JARS` environment variable. For example, using `maven-surefire-plugin`:
+
+```
+<plugin>
+    <groupId>org.apache.maven.plugins</groupId>
+    <artifactId>maven-surefire-plugin</artifactId>
+    <version>3.0.0-M4</version>
+    <configuration>
+        <reuseForks>false</reuseForks>
+        <environmentVariables>
+            ...
+            <LOCAL_JARS>../target/atlassian-bitbucket-server-integration.hpi</LOCAL_JARS>
+            ...
+        </environmentVariables>
+    </configuration>
+</plugin>
+```
+
+To build the `.hpi` file, build the plugin from the project root folder:
+
+```
+mvn clean package -DskipTests
+```
+
+---
+If you see the following error when running the tests, it means you haven't built the plugin:
+
+```
+java.lang.IllegalArgumentException: Unable to honor LOCAL_JARS environment variable
+
+	at org.jenkinsci.test.acceptance.update_center.LocalOverrideUpdateCenterMetadataDecoratorImpl.decorate(LocalOverrideUpdateCenterMetadataDecoratorImpl.java:83)
+	at org.jenkinsci.test.acceptance.update_center.CachedUpdateCenterMetadataLoader.get(CachedUpdateCenterMetadataLoader.java:48)
+	at org.jenkinsci.test.acceptance.update_center.MockUpdateCenter.ensureRunning(MockUpdateCenter.java:100)
+	at org.jenkinsci.test.acceptance.po.PluginManager.checkForUpdates(PluginManager.java:87)
+	at org.jenkinsci.test.acceptance.po.PluginManager.installPlugins(PluginManager.java:182)
+...
+Caused by: java.lang.IllegalArgumentException: Plugin file does not exist: <path-to-project-root->/acceptance-tests/../target/atlassian-bitbucket-server-integration.hpi
+	at org.jenkinsci.test.acceptance.update_center.LocalOverrideUpdateCenterMetadataDecoratorImpl.override(LocalOverrideUpdateCenterMetadataDecoratorImpl.java:91)
+	at org.jenkinsci.test.acceptance.update_center.LocalOverrideUpdateCenterMetadataDecoratorImpl.decorate(LocalOverrideUpdateCenterMetadataDecoratorImpl.java:81)
+	... 27 more
+```
+---
+
+Then run the tests from inside the `acceptance-tests` folder:
+
+```
+cd acceptance-tests
+mvn test
+```
+
+This will run all the tests.
+
+You can also run individual tests inside Intellij as usual.
```

Comment:
I just remembered (again) this is a public open-source repo, so I should probably change this to "inside your IDE of choice".